### DISCUSSION TITLE
Are Building and Testing MOOSE required every time I update Conda and MOOSE repository?

### DISCUSSION THREAD

**Discussion by kurtenkera** [idaholab/moose#26948] [Q&A General]
I'm trying to develop a MOOSE-app at the moment. Every week I am updating the MOOSE repository and Conda by following the steps seen below:

![image](https://github.com/idaholab/moose/assets/127472392/fc1f7ccb-25d1-42f5-b66c-3a11e0c19e81)

I then return to my application, and run the following to re-build and test it:

```
cd ~/projects/YourAppName
make clobberall
make -j 6
./run_tests -j 6
```
This is my question: Do I need to re-build and test MOOSE itself by following the steps below every time I update the MOOSE repository and Conda? Or is it enough simply re-build and test my application as seen above? In other words, are we only required to execute the build and testing of MOOSE below when we first download the MOOSE repository? I need to know as building and testing MOOSE using the steps below is very time consuming, and I'd rather not do it every time I update Conda and the MOOSE repository! Thanks in advance!

![image](https://github.com/idaholab/moose/assets/127472392/f921b078-6cc6-419c-8617-8840af5ca724)

**Comment by dschwen** [comment]
You do not need to jump through all those hoops every week. Pay attention of the version of the moose conda packages is actually increased. If not, there's no need to do a clobberall. Testing is up to you. It should not be necessary, but if you want to run a simulation and want to rely on the results, it is recommended that you run the test suite to see if there are any problems with your installation.
